### PR TITLE
zone_weather slim down / optimize for speed

### PR DESCRIPTION
**Pull request by Kreidos**
No longer can zone_weather be the butt of jokes. She's lean and fast and remembers how you laughed when she was big; so she's not giving you her number.

In seriousness, zone_weather is big. 237,000 rows of daily weather data that takes a long time to import, and a lot of space to store. It's the largest DB table by a wide margin. This switches the inefficient existing storage to just one neatly packed bit pattern per zone. Those interested in technical specs can see the comment above loadZoneWeather() where it's detailed for you.

### Gains
```
SQL File Size:              83% reduction
SQL Import time:            97% reduction
Server weather load:        55% reduction
Raw table Size:             40% reduction
```
This will be especially welcome to those who aren't running on high end hardware, or those trying to run on Raspberry Pi's where SQL transaction IO can be limiting.

### Considerations
The table will no longer be in human-readable form, though I would argue it was far from human-friendly before, and the weather table has generally gone untouched for quite some time so I consider it essentially static. That said, the fixed-length data format was designed to be easy to work with, and a tool or script could be made to make editing zones easier than was even possible previously.

The new weather blobs ~~live in a new table which~~ needs to be imported or the server will display errors that the user should "Import zone_weather.sql". I don't believe any migrate step is needed at this time, as the table is new and shares nothing in common with the old one. 

**_I affirm:_**
- [x] that I agree to Project Topaz's [Limited Contributor License Agreement](http://project-topaz.com/blob/release/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I've _tested my code_ since the last commit in the PR, and will test after any later commits

### CODE REVIEW

**Comment by Apocarypse** [comment]
RIP thicc zone_weather
hello slimfast weather!

File: src/map/zone.cpp [hunk 1]
```diff
@@ -290,37 +290,44 @@ void CZone::LoadZoneLines()
...
     if (ret != SQL_ERROR && Sql_NumRows(SqlHandle) != 0)
     {
-        while (Sql_NextRow(SqlHandle) == SQL_SUCCESS)
-        {
-            m_WeatherVector.insert(std::make_pair((uint16)Sql_GetUIntData(SqlHandle, 0), zoneWeather_t(Sql_GetIntData(SqlHandle, 1),
-                Sql_GetIntData(SqlHandle, 2), Sql_GetIntData(SqlHandle, 3))));
-        }
+            Sql_NextRow(SqlHandle);
+            auto weatherBlob = reinterpret_cast<uint16*>(Sql_GetData(SqlHandle, 0));
+            for(uint16 i=1; i<WEATHER_CYCLE; i++)
```
Comment (zach2good) [inline src/map/zone.cpp]: Would like some spaces in here:
`if (...`
`i = 1;`
etc.

File: src/map/zone.cpp [hunk 2]
```diff
@@ -290,37 +290,44 @@ void CZone::LoadZoneLines()
     }
 }
 
-/************************************************************************
-*                                                                       *
-*  Загружаем параметры погоды                                           *
-*                                                                       *
-************************************************************************/
+/*************************************************************************
```
Comment (zach2good) [inline src/map/zone.cpp:293]: Thanks for the excellent comment explaining the bit shifting witchcraft :)

**Review by zach2good — APPROVED** [review]
LGTM! Just some very minor styling requests
image
_gotta go fast_

File: src/map/zone.cpp (hunk 1 above)
Comment (Kreidos) [reply src/map/zone.cpp]: Updated to give you some space!

**Comment by TeoTwawki** [comment]
> _gotta go fast_

**Review by cocosolos — APPROVED** [review]

**Review by ibm2431 — APPROVED** [review]
Straight into release with you!

**Pull request merged**